### DISCUSSION TITLE
Suggested fixes for 5.0 release

### DISCUSSION THREAD

**Discussion by efeoli** [timschofield/webERP#212] [Problems / Bugs]
Hello Tim,
as reported earlier webERP can be now installed and run in PHP8 environment, Also prior databases may be imported and run.

Testing the latter I have found the following:
List of webERP faulty functions
- [ ] Utilities module will not create a new company
- [ ] For customer’s modify sales orders, the screen that follows will not print pick lists.
- [ ] The WO order ready to produce in manufacturing module is still absent. However, this function may be reached form “Commonly used scripts” kept in the imported database and it is fully functional at  [https://kwa.bioacyl.com/webERP/WOCanBeProducedNow.php ](https://kwa.bioacyl.com/webERP/WOCanBeProducedNow.ph)This issue then, would not be crucial for me personally, but would confuse newcomers and it is a pity it being functional and so valuable for manufacturing enterprises but not reachable, simply for a menu problem
- [ ] For select a work order in manufacturing module, screen that follows work order can’t be edited and it doesn’t print. It may be edited in composite orders, but only for the second entry. Batches tab is necessary for first entry. Work order doesn’t print either in “Select a Work Order” tab. This really is problematic for good manufacturing regulated industries 
- [ ] Imported reports do reach the report builder tool, but do not show in the respective module.
- [ ] Currencies entry and maintenance in setup module is not working and affects any transaction involving currencies 

- [ ] Enter counter sales doesn’t find item to be entered when searched by item code
- [ ] Sales graph opens only a thumb sized graph and not the full image and so Graph of Account Transactions
- [ ] Session duration is quite short and despite of editing session duration in config.php as set previously, continues to logout quite fast. The system will logout even if one is working 
- [ ] Asset manager email reminder tab, is not working
- [ ] There is no PDF printing for Petty Cash tab General Report and PC Expense General Report
- [ ] Most of GL reports of balance/statements are not printing PDFs
- [ ] Mail Inventory Valuation Report generates TCPDF error: Unable to create output file companies/bioacylc_erp/reports/InventoryReport.pdf 
- [ ] Geocode Maintenance isn’t working. It generates this notice: Customer Code: 3HCRurl not loading 
- [ ] In Select pending sales orders this error is shown: 
ERROR Report : The date does not appear to be in a valid format. The date being converted from SQL format was:

I will send Currencies screenshot separately.
Enrique

**Comment by efeoli** [comment]
![Captura de pantalla_10-11-2024_11016_kwa bioacyl com](https://github.com/user-attachments/assets/9ad12264-9ad8-4836-a673-b638164ceec5)

**Reply by efeoli** [reply]
I hadn’t noticed the Latest News instructions for timeout, so please disregard this point, since it is solved.

**Reply by dalers** [reply]
Very impressed @efeoli!  Thank you for advance notice of manufacuturing issues (a good manufacturing workflow is very important for me).

Fwiw, I have encouring difficulties in the manual (selecting pages to view on the front page, and missing images on other pages...) and will create a placeholder issue now and add details and screenshots later.

**Reply by timschofield** [reply]
Hi Dale, I think thjese manual images should now be fixed in the following commit - https://github.com/timschofield/webERP/commit/c8bd38a9e8c85928c99174c93279b5c9527e9f56 
You may need to do a ctrl-F5 to force the css to reload.

Thanks
Tim

**Comment by timschofield** [comment]
Fantastic Enrique thanks very much for the testing. I will get back you as
I make my way through the list.

Thanks for your hard work,
Tim

On Mon, 11 Nov 2024 at 16:04, efeoli ***@***.***> wrote:

> Hello Tim,
> as reported earlier webERP can be now installed and run in PHP8
> environment, Also prior databases may be imported and run.
>
> Testing the latter I have found the following:
> List of webERP faulty functions
>
>    -
>
>    Utilities module will not create a new company
>    -
>
>    For customer’s modify sales orders, the screen that follows will not
>    print pick lists.
>    -
>
>    The WO order ready to produce in manufacturing module is still absent.
>    However, this function may be reached form “Commonly used scripts” kept in
>    the imported database and it is fully functional at https://kwa.bioacyl.com/webERP/WOCanBeProducedNow.php
>    <https://kwa.bioacyl.com/webERP/WOCanBeProducedNow.ph>This issue then,
>    would not be crucial for me personally, but would confuse newcomers and it
>    is a pity it being functional and so valuable for manufacturing enterprises
>    but not reachable, simply for a menu problem
>    -
>
>    For select a work order in manufacturing module, screen that follows
>    work order can’t be edited and it doesn’t print. It may be edited in
>    composite orders, but only for the second entry. Batches tab is necessary
>    for first entry. Work order doesn’t print either in “Select a Work Order”
>    tab. This really is problematic for good manufacturing regulated industries
>    -
>
>    Imported reports do reach the report builder tool, but do not show in
>    the respective module.
>    -
>
>    Currencies entry and maintenance in setup module is not working and
>    affects any transaction involving currencies
>    -
>
>    Enter counter sales doesn’t find item to be entered when searched by
>    item code
>    -
>
>    Sales graph opens only a thumb sized graph and not the full image and
>    so Graph of Account Transactions
>    -
>
>    Session duration is quite short and despite of editing session
>    duration in config.php as set previously, continues to logout quite fast.
>    The system will logout even if one is working
>    -
>
>    Asset manager email reminder tab, is not working
>    -
>
>    There is no PDF printing for Petty Cash tab General Report and PC
>    Expense General Report
>    -
>
>    Most of GL reports of balance/statements are not printing PDFs
>    -
>
>    Mail Inventory Valuation Report generates TCPDF error: Unable to
>    create output file companies/bioacylc_erp/reports/InventoryReport.pdf
>    -
>
>    Geocode Maintenance isn’t working. It generates this notice: Customer
>    Code: 3HCRurl not loading
>    -
>
>    In Select pending sales orders this error is shown:
>    ERROR Report : The date does not appear to be in a valid format. The
>    date being converted from SQL format was:
>
> I will send Currencies screenshot separately.
> Enrique
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/timschofield/webERP/discussions/212>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAL6LESNDD4OLBULALA4MD2ADIRDAVCNFSM6AAAAABRSEW4YCVHI2DSMVQWIX3LMV43ERDJONRXK43TNFXW4OZXGQ3DKMRSGA>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
www.weberp.org <http://www.weberpafrica.com>
@TimSchofield2
Blog: https://kwamoja.home.blog/

**Reply by efeoli** [reply]
My pleasure Tim and glad 😀 to help.
Enrique

**Comment by timschofield** [comment]
I have converted these to separate issues on Github.

The sales graph issue
(https://github.com/timschofield/webERP/issues/228) has been solved
with commit https://github.com/timschofield/webERP/commit/cdcd86a7e1b6f251decf40c69beac48ac8690e54

Thanks
Tim

On Mon, 11 Nov 2024 at 16:04, efeoli ***@***.***> wrote:
>
> Hello Tim,
> as reported earlier webERP can be now installed and run in PHP8 environment, Also prior databases may be imported and run.
>
> Testing the latter I have found the following:
> List of webERP faulty functions
>
> Utilities module will not create a new company
>
> For customer’s modify sales orders, the screen that follows will not print pick lists.
>
> The WO order ready to produce in manufacturing module is still absent. However, this function may be reached form “Commonly used scripts” kept in the imported database and it is fully functional at https://kwa.bioacyl.com/webERP/WOCanBeProducedNow.php This issue then, would not be crucial for me personally, but would confuse newcomers and it is a pity it being functional and so valuable for manufacturing enterprises but not reachable, simply for a menu problem
>
> For select a work order in manufacturing module, screen that follows work order can’t be edited and it doesn’t print. It may be edited in composite orders, but only for the second entry. Batches tab is necessary for first entry. Work order doesn’t print either in “Select a Work Order” tab. This really is problematic for good manufacturing regulated industries
>
> Imported reports do reach the report builder tool, but do not show in the respective module.
>
> Currencies entry and maintenance in setup module is not working and affects any transaction involving currencies
>
> Enter counter sales doesn’t find item to be entered when searched by item code
>
> Sales graph opens only a thumb sized graph and not the full image and so Graph of Account Transactions
>
> Session duration is quite short and despite of editing session duration in config.php as set previously, continues to logout quite fast. The system will logout even if one is working
>
> Asset manager email reminder tab, is not working
>
> There is no PDF printing for Petty Cash tab General Report and PC Expense General Report
>
> Most of GL reports of balance/statements are not printing PDFs
>
> Mail Inventory Valuation Report generates TCPDF error: Unable to create output file companies/bioacylc_erp/reports/InventoryReport.pdf
>
> Geocode Maintenance isn’t working. It generates this notice: Customer Code: 3HCRurl not loading
>
> In Select pending sales orders this error is shown:
> ERROR Report : The date does not appear to be in a valid format. The date being converted from SQL format was:
>
> I will send Currencies screenshot separately.
> Enrique
>
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>



-- 
www.weberp.org
@TimSchofield2
Blog: https://kwamoja.home.blog/

**Reply by efeoli** [reply]
Still getting as per

![Copia de Captura de pantalla_11-11-2024_16716_kwa bioacyl com](https://github.com/user-attachments/assets/407f62c2-43e8-48b4-bd16-406f262154bf)

**Reply by efeoli** [reply]
This is after opening the thumb
![salesgraph (1)](https://github.com/user-attachments/assets/187acea8-5130-4f1d-aff4-81eece5dc00d)

**Comment by timschofield** [comment]
You will need to force the browser to update to the latest css rather than
use the cache - Ctrl-f5 in most browsers.

Thanks
Tim

On Mon, 11 Nov 2024 at 22:19, efeoli ***@***.***> wrote:

> Still getting as per
>
> Copia.de.Captura.de.pantalla_11-11-2024_16716_kwa.bioacyl.com.jpeg (view
> on web)
> <https://github.com/user-attachments/assets/407f62c2-43e8-48b4-bd16-406f262154bf>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/timschofield/webERP/discussions/212#discussioncomment-11219349>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAL6LHKT22WXACWLAHDURT2AEUOXAVCNFSM6AAAAABRSEW4YCVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCMRRHEZTIOI>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
www.weberp.org <http://www.weberpafrica.com>
@TimSchofield2
Blog: https://kwamoja.home.blog/

**Comment by timschofield** [comment]
Did you try forcing a reload of the css with a ctrl-f5 as I suggested?

Tim

On Mon, 11 Nov 2024 at 22:27, efeoli ***@***.***> wrote:
>
> This is after opening the thumb
> salesgraph.1.png (view on web)
>
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you commented.Message ID: ***@***.***>



-- 
www.weberp.org
@TimSchofield2
Blog: https://kwamoja.home.blog/

**Reply by efeoli** [reply]
Yes, but caché seems to be stuck in several browsers. Sometimes only renews after several hours.
Enrique

**Reply by efeoli** [reply]
Hi again,
sales graph behaving the same this morning.
Enrique

**Reply by efeoli** [reply]
With last commit image seems to be absent 
![Copia de Captura de pantalla_11-11-2024_16716_kwa bioacyl com](https://github.com/user-attachments/assets/5701955d-24fc-4348-82f2-2713a5a27b4a)

**Reply by efeoli** [reply]
![Captura de pantalla_12-11-2024_11849_kwa bioacyl com](https://github.com/user-attachments/assets/992a41e0-486b-4180-9181-b4c731904e19)

**Reply by efeoli** [reply]
Sales graph now working perfectly Tim!

**Comment by timschofield** [comment]
OK, I think I have finally solved this. The problem seems to be when webERP
isn't in the web root. Can you try the latest version I have just
committed? It works on my local system and on the demo.

Thanks
Tim

On Tue, 12 Nov 2024 at 17:17, efeoli ***@***.***> wrote:

> Captura.de.pantalla_12-11-2024_11849_kwa.bioacyl.com.jpeg (view on web)
> <https://github.com/user-attachments/assets/992a41e0-486b-4180-9181-b4c731904e19>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/timschofield/webERP/discussions/212#discussioncomment-11229643>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAL6LA53LCNPT24QIUMXML2AIZZZAVCNFSM6AAAAABRSEW4YCVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCMRSHE3DIMY>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
www.weberp.org <http://www.weberpafrica.com>
@TimSchofield2
Blog: https://kwamoja.home.blog/

**Reply by efeoli** [reply]
Yes testing was with 8436 commit.

**Comment by timschofield** [comment]
Good stuff. I've had a lot of personal stuff to do today, tomorrow I will
make greater inroads into your list.

Thanks for your work on this,
Tim

On Tue, 12 Nov 2024 at 19:43, efeoli ***@***.***> wrote:

> Sales graph now working perfectly Tim!
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/timschofield/webERP/discussions/212#discussioncomment-11231362>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAL6LCHOLJONZJA3TJSFUL2AJK6BAVCNFSM6AAAAABRSEW4YCVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCMRTGEZTMMQ>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
www.weberp.org <http://www.weberpafrica.com>
@TimSchofield2
Blog: https://kwamoja.home.blog/

**Reply by efeoli** [reply]
webERP is a top ERP and I am for it to be where it belongs.

**Reply by efeoli** [reply]
Hi Tim,
despite recent commit WO ready to produce, still not showing and forcing cache erasing.
Enrique

**Reply by efeoli** [reply]
Thank you for explaining. I just overwrote the new commit. I will do a new install in a while after some work to be done.
Enrique

**Comment by timschofield** [comment]
Hi Enrique, what I have fixed is that the demo data has that link in it. If
you did a new install you should see them.

I think I need to do a database update file for the case of those people
importing an old database. The trouble with that is all new install will
already contain it. Trouble is new installs will already have it in, but
they will just ignore it.

Thanks
Tim

On Wed, 13 Nov 2024 at 21:06, efeoli ***@***.***> wrote:

> Hi Tim,
> despite recent commit WO ready to produce, still not showing and forcing
> cache erasing.
> Enrique
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/timschofield/webERP/discussions/212#discussioncomment-11246346>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAL6LH35L5D5OKPV4SY5QD2AO5NHAVCNFSM6AAAAABRSEW4YCVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCMRUGYZTINQ>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
www.weberp.org <http://www.weberpafrica.com>
@TimSchofield2
Blog: https://kwamoja.home.blog/

**Reply by efeoli** [reply]
Thank you for explaining. I just overwrote the new commit. I will do a new install in a while after some work to be done.
Enrique

**Comment by timschofield** [comment]
Thanks, I would be grateful for you checking it works

On Wed, 13 Nov 2024 at 21:29, efeoli ***@***.***> wrote:

> Thank you for explaining. I just overwrote the new commit. I will do a new
> install in a while after some work to be done.
> Enrique
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/timschofield/webERP/discussions/212#discussioncomment-11246518>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAL6LC357FLM7EJFIIZYTT2APACZAVCNFSM6AAAAABRSEW4YCVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCMRUGY2TCOA>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
www.weberp.org <http://www.weberpafrica.com>
@TimSchofield2
Blog: https://kwamoja.home.blog/

**Reply by efeoli** [reply]
The fresh install does not show WO ready to produce.
![Captura de pantalla_13-11-2024_161853_kwa bioacyl com](https://github.com/user-attachments/assets/99dbcd0c-13eb-49bb-a5a9-b53498264a5e)
Enrique

**Comment by timschofield** [comment]
Was this with the demo data, or just the structure?

On Wed, 13 Nov 2024 at 22:24, efeoli ***@***.***> wrote:

> The fresh install does not show WO ready to produce.
> Captura.de.pantalla_13-11-2024_161853_kwa.bioacyl.com.jpeg (view on web)
> <https://github.com/user-attachments/assets/99dbcd0c-13eb-49bb-a5a9-b53498264a5e>
> Enrique
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/timschofield/webERP/discussions/212#discussioncomment-11246901>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAL6LENJMWQ7VULY5KOANL2APGQTAVCNFSM6AAAAABRSEW4YCVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCMRUGY4TAMI>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
www.weberp.org <http://www.weberpafrica.com>
@TimSchofield2
Blog: https://kwamoja.home.blog/

**Reply by efeoli** [reply]
With the demo data.

**Reply by efeoli** [reply]
Also something went wrong with the demo changes. When you import database from 4
![Captura de pantalla_13-11-2024_171535_kwa bioacyl com](https://github.com/user-attachments/assets/18bbe013-f349-45d6-a9a3-a22b662fc0c9)
.15.2 now all execution actions are missing. This is a capture from Customer Transaction Inquiries.

Enrique

**Reply by efeoli** [reply]
You cannot print, issue a credit note , visualize HTML or allocate.

**Reply by efeoli** [reply]
Happens with fresh start demo data.
![Captura de pantalla_13-11-2024_175443_kwa bioacyl com](https://github.com/user-attachments/assets/2422c7bb-aa9b-4100-9882-6477cae6828d)
